### PR TITLE
Convert controllers to native classes part one

### LINKED ISSUE #721: Rewrite controllers in native class (Part 1 of 2)
## Background

Over the next month or so, we'll upgrade `ember-source` from `v3.20` to `v3.24` and ensure that we follow recommended practices in Ember Octane.

## Description

In #720, I added unit tests for controllers. These tests, in particular, assert how we transform and sort data before we pass them to a route template.

Now that we have tests, let's rewrite the controllers to use native class.

## TODOs

Please update the following controllers that have a low risk of breaking the app. We can rely on unit tests to say that the app functions as intended.

- [ ] Rewrite `application` controller in native class.
- [ ] Rewrite `ember-users` controller in native class.
- [ ] Rewrite `sponsors` controller in native class.
- [ ] Rewrite `team` controller in native class.
- [ ] Rewrite `releases.index` controller in native class.

## References

If you are unsure about the syntax for controllers in native class, please have a look at the following links in the Ember Guides:

- https://guides.emberjs.com/release/routing/controllers/
- https://guides.emberjs.com/release/routing/query-params/

### PR DESCRIPTION
**Pull request by sukima**
This should address the controllers in issue #721

Closes #721

### CODE REVIEW

**Comment by sukima** [comment]
I'm having test failures on `master` date based errors. I do not know how to fix.

This PR requires code review now that tests are not trusted due to master failing as well.

File: app/controllers/application.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { infoLinks } from 'ember-styleguide/constants/es-footer';
 import styleguideLinks from 'ember-styleguide/constants/links';
 import replaceLinks from 'ember-website/utils/replace-links';
 
-export default Controller.extend({
-  links: replaceLinks(styleguideLinks),
-  infoLinks: replaceLinks(infoLinks),
-});
+export default class Application extends Controller {
```
Comment (ijlee2) [inline app/controllers/application.js:6]: Can you update this line (similarly, in other controllers) by adding the suffix `Controller`? I'd like the code to follow what an Ember CLI-generated file would show.

```suggestion
export default class ApplicationController extends Controller {
```

File: app/controllers/ember-users.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
-  sortingKey: Object.freeze(['featured:desc', 'added']),
-  sortedModel: sort('model', 'sortingKey'),
-});
+  get sortedModel() {
+    return this.model.toArray().sort((a, b) => {
+      let featuredComp = (a.featured < b.featured) * 2 - 1;
+      let addedComp = a.added - b.added;
+      return -featuredComp || addedComp; // featured DESC, added ASC
+    });
+  }
```
Comment (ijlee2) [inline app/controllers/ember-users.js:13]: I agree that we can use JavaScript's `.sort()` to sort the data. However, to keep the pull request's scope small for review, can you use `@sort` instead?

```suggestion
  sortingKey = Object.freeze(['featured:desc', 'added']);
  /* eslint-disable-next-line ... */
  @sort('model', 'sortingKey') sortedModel;
```

I think making fewer changes in one pull request will help with tests passing.

File: app/controllers/releases/index.js [hunk 3]
```diff
@@ -1,10 +1,9 @@
-import { computed } from '@ember/object';
 import Controller from '@ember/controller';
 
-export default Controller.extend({
-  emberBetaProject: computed('model.@each.{name,channel}', function () {
+export default class ReleasesIndex extends Controller {
+  get emberBetaProject() {
```
Comment (ijlee2) [inline app/controllers/releases/index.js:4]: Looks good. 👍

File: app/controllers/team.js [hunk 4]
```diff
@@ -1,21 +1,28 @@
...
-  return computed('sortedModel.[]', function () {
-    return this.sortedModel.filter((member) => member.teams.includes(team));
-  });
+  return (target, key) => {
+    Object.defineProperty(target, key, {
+      get() {
+        return this.sortedModel.filter((member) => member.teams.includes(team));
+      },
+    });
+  };
```
Comment (ijlee2) [inline app/controllers/team.js:10]: I thought your decorator approach was interesting. At a later point, I think we'll end up making a utility function that filters `this.model`, but we can keep this code around for now.

**Review by ijlee2 — CHANGES_REQUESTED** [review]
@sukima Thank you for making your 1st contribution to the `ember-website` repo.

To help me review quickly and help you get the tests to pass, can you revert how you updated `sort` and use `@sort` instead? Afterward, if you will also update the class name by appending the suffix `Controller`, that'd be great!

File: app/controllers/application.js (hunk 1 above)
Comment (sukima) [reply app/controllers/application.js:6]: Interesting, I asked my team about this a few months ago and the consensus was that the prefix was redundant. 🤷

**Comment by sukima** [comment]
Changes completed

**Review by ijlee2 — APPROVED** [review]
Thank you for addressing my feedback. I especially appreciated that you put extra effort into keeping the commit history relevant. 💯

**Pull request merged**